features/default_backend.feature

```diff
@@ -1,49 +1,47 @@
 @sig-network @conformance @release-1.19
 Feature: Default backend
-  An Ingress with no rules sends all traffic to a single default backend.
-  The default backend is typically a configuration option of the
-  Ingress controller and is not specified in your Ingress resources.
+
+  An Ingress with no rules sends all traffic to the single default backend.
+  The default backend is part of the Ingress resource spec field `defaultBackend`.
 
   If none of the hosts or paths match the HTTP request in the
   Ingress objects, the traffic is routed to your default backend.
 
-    Rules:
-    - Response status code is 404.
-    - Response body contains arbitrary text.
-
-        Scenario: Sample scenario for code-generation -- Ingress with host and no backend serviceName
-            Given a new random namespace
-              And reading Ingress from manifest "scenarios/001/ing.yaml"
-             Then creating Ingress from manifest returns an error message containing "spec.rules[0].http.paths[0].backend.serviceName: Required value"
-
-        Scenario: Sample scenario for code-generation -- Ingress with host and invalid backend
-            Given a new random namespace
-              And reading Ingress from manifest "scenarios/002/ing.yaml"
-              And creating Ingress from manifest
-             When The ingress status shows the IP address or FQDN where is exposed
-              And Header "Host" with value "foo.bar"
-              And Send HTTP request with method "GET"
-             Then Response status code is 404
+  Background:
+    Given a new random namespace
+    Given an Ingress resource named "default-backend" with this spec:
+      """
+      spec:
+        defaultBackend:
+          service:
+            name: echo-service
+            port:
+              number: 80
+      """
+    Then The Ingress status shows the IP address or FQDN where it is exposed
 
-        Scenario: Sample scenario for code-generation -- Ingress should return 404 for paths with an invalid backend serviceName
-            Given a new random namespace
-              And reading Ingress from manifest "scenarios/003/ing.yaml"
-              And creating Ingress from manifest
-             When The ingress status shows the IP address or FQDN where is exposed
-              And Header "Host" with value "foo.bar"
-             Then Send HTTP request with <path> and <method> checking response status code is 404:
-                  |  path   | method  |
-                  | /test   | GET     |
-                  | /       | POST    |
-                  | /       | PUT     |
-                  | /       | DELETE  |
-                  | /       | GET     |
+  Scenario Outline: An Ingress with no rules should send all requests to the default backend
+    When I send a "<method>" request to http://"<host>"/"<path>"
```

Comment:
How will we setup the Services/Pods backing these Ingresses? Will there be some common resources that can be referenced by all tests?

Reply:
Yes, https://github.com/kubernetes-sigs/ingress-controller-conformance/blob/master/tools/echoserver.go or some similar existing image already being used in k/k